src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -296,8 +297,9 @@ internal interface ISyntaxFactsService : ILanguageService
         bool IsQueryKeyword(SyntaxToken token);
         bool IsThrowExpression(SyntaxNode node);
         bool IsElementAccessExpression(SyntaxNode node);
+        bool IsPartOfPropertyDeclarationHeader(SyntaxNode node);
```

Comment:
can you move next to IsOnTypeHeader.  Can we also name the same and make the signatures the same?

Reply:
Will move.
As for name, the name has different style because it doesn't take Location but a node so I wanted to explicitly differentiate them. As for changing the signature (and then the name) that would obfuscate stuff somehow. Essentially I'd get a Node in the Service -> took it's .Span.Begin, pass it to IsOnDeclarationHeader -> in it retrieved Token & potentially it's ancestors node again (the one I already had) & ...

Reply:
Since I expect more IsPartOfXXXHeader methods I'd be against creating obfuscation just for the sake of being totaly consistent with one already existing method.

Reply:
> As for name, the name has different style because it doesn't take Location but a node

Why? :)   Can they take the same thing?  

> As for changing the signature (and then the name) that would obfuscate stuff somehow. Essentially I'd get a Node in the Service -> took it's .Span.Begin, pass it to IsOnDeclarationHeader -> in it retrieved Token & potentially it's ancestors node again (the one I already had) & ...

Doesn't seem obfuscated to me :)

It means each guy is just checking things in a very similar manner.  Basically we're consistent in that we have a common way to say: Given this position in the tree, are you in this type of header?  Nice, simple and conceptually clean.

Reply:
> Since I expect more IsPartOfXXXHeader methods I'd be against creating obfuscation just for the sake of being totaly consistent with one already existing method.

If you believe that, then i would prefer IsInTypeHeader to move to IsPartOfTypeHeader :)  But, IMO, the IsInTypeHeader is really clear and simple to understand/implement.  Trust me when i say that when a care moves, we likely find the token under it (and the ancestors) many dozens of times.  that's not a hotspot for us, and there's little motivation to try to prevent it from happen *across* services like this.

Reply:
> Given this position in the tree, are you in this type of header? Nice, simple and conceptually clean.

But what the service really wants is asking "Is this node in header of this type"? Not "is this position in header of this type". I know that you can answer the first question through the second one but it's one step (and I mean conceptually not performance-wise) that isn't necessary.

Reply:
> But what the service really wants is asking "Is this node in header of this type"? 

I guess my point is this: these are different services.  It's ok with me that the way one service has designed the shape of questions/answers isn't perfectly aligned with what another service/feature exactly has.

Reply:
I would agree with that _if_ the SyntaxFactsService didn't already answer questions about node and was purely location based. But that's not the case, there're plenty IsXXX(SyntaxNode node) methods.

I'd also agree if we could see a usecase for IsOnXXXHeader(int location, …) that couldn't be (better) served with IsPartOfXXXHeader(SyntaxNode node) but that's also IMO not the case.

Reply:
> IsInTypeHeader to move to IsPartOfTypeHeader :) 

That also can't happen very easily / without clouding the way the refactorings that use it work. It's used for "generate within type" refactorings that inherently operate on locations (in header || between types) rather than only nodes.

Reply:
I _really_ don't see the benefit of unification here, to be honest.

Reply:
I have renamed the helper to convey they're more related & to make the name clearer. I still think further merging is not preferable. Will resolve for now unless you have a really strong prefferance for merging them.

Reply:
i do have a pretty strong preference here.  They're basically intended for the exact same scenarios.  A user has their cursor somewhere and we want to ask "are they are hte header of something".  It doesn't make much sense for me to not have them look and behave in a consistent fashion.